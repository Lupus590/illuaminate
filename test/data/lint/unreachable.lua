-- config: (lint (only (control:unreachable)))

local f = ...

local function _()
  do return end

  print()
end

local function _()
  if f() then
    return 1
  else
    return 2
  end

  if f() then
    return 1
  else
    return 2
  end

  return
end

local function _()
  while true do end

  return
end

local function _()
  while true do
    if f() then break end
  end

  return -- ok
end

local function _()
  repeat
    return
  until f()
end


local function _()
  while true do
    do break end

    print()
  end
end

local function _()
  if true then
  elseif f() then
    print() -- ok
  elseif false then
    print()
  end

end

do return end
print()
