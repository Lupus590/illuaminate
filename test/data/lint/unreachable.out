unreachable.lua:[8:3-8:10]: Unreachable code [control:unreachable]
   │
 8 │   print()
   │   ^^^^^^^
unreachable.lua:[18:3-22:6]: Unreachable code [control:unreachable]
    │
 18 │   if f() then
    │   ^^^^^^^^^^^
unreachable.lua:[24:3-24:9]: Unreachable code [control:unreachable]
    │
 24 │   return
    │   ^^^^^^
unreachable.lua:[30:3-30:9]: Unreachable code [control:unreachable]
    │
 30 │   return
    │   ^^^^^^
unreachable.lua:[42:3-44:12]: Loop is executed at most once. [control:loop-once]
    │
 42 │   repeat
    │   ^^^^^^
unreachable.lua:[44:9-44:12]: Unreachable code [control:unreachable]
    │
 44 │   until f()
    │         ^^^
unreachable.lua:[52:5-52:12]: Unreachable code [control:unreachable]
    │
 52 │     print()
    │     ^^^^^^^
unreachable.lua:[58:10-58:13]: Unreachable code [control:unreachable]
    │
 58 │   elseif f() then
    │          ^^^
unreachable.lua:[61:5-61:12]: Unreachable code [control:unreachable]
    │
 61 │     print()
    │     ^^^^^^^
unreachable.lua:[67:1-67:8]: Unreachable code [control:unreachable]
    │
 67 │ print()
    │ ^^^^^^^
No errors and 10 warnings
